cibuildwheel/util.py

```diff
@@ -562,6 +565,46 @@ def virtualenv(
     return env
 
 
+@overload
+def find_abi3_wheel(wheels: Sequence[Path], identifier: str) -> Optional[Path]:
+    ...
+
+
+@overload
+def find_abi3_wheel(wheels: Sequence[PurePath], identifier: str) -> Optional[PurePath]:
+    ...
+
+
+def find_abi3_wheel(
+    wheels: Sequence[Union[Path, PurePath]], identifier: str
```

Comment:
Couldn't you use `TypeVar("P", bound=PurePath)` instead of the overload? Maybe even `Traversable`, though I forget if you can bind to something structurally subclassing, I think you can.

Reply:
I was not very keen on what I came up with but couldn't find easily other ways to do that. I'm not very comfortable with  typing yet except for basic stuff but I'll try to look into your proposals over the week-end. If you have something you want to commit in the meantime, please do.

Reply:
I tried what I was thinking in ba9a91e.

Reply:
Thanks @henryiii, looks much better this way and I learned something.

Reply:
Thanks @henryiii, looks much better this way and I learned something.